consumer-test/runTests.sh

```diff
@@ -0,0 +1,14 @@
+#!/bin/bash -e
+
+LIBRARY_VERSION_UNDER_TEST=$(git log --format="%H" -n 1)
+REPOSITORY_UNDER_TEST=${TRAVIS_REPO_SLUG:-SAP/jenkins-library}
+
+rm -rf workspace
+git clone -b consumer-test https://github.com/sap/cloud-s4-sdk-book workspace
+cp -f jenkins.yml workspace
+cd workspace
+sed -i -e "s:__REPO_SLUG__:${REPOSITORY_UNDER_TEST}:g" jenkins.yml
+echo "@Library(\"piper-library-os@$LIBRARY_VERSION_UNDER_TEST\") _" | cat - Jenkinsfile > temp && mv temp Jenkinsfile
+git commit --all --author="piper-testing-bot <null@null.com>" --message="Set piper lib version for test"
```

Comment:
Is the git commit here really needed? The Jenkinsfile is contained in the workspace and the workspace is handed over into the docker sesssion later.

Reply:
I introduced the commit because `artifactSetVersion` fails if the git tree is dirty. What is the better alternative?